File: JavaLibrary/src/main/java/com/flat/simplex/parser/logic/line/call/CallField.java
package com.flat.simplex.parser.logic.line.call;

import com.flat.simplex.lexer.Token;
import com.flat.simplex.parser.logic.Block;
import com.flat.simplex.parser.logic.Field;
import com.flat.simplex.parser.logic.error.Error;

public class CallField extends LineCall {

    private Field field;

    public CallField(Block parent, Token token) {
        super(parent, token, Type.Field);
    }

    @Override
    public void load() {
        if (getPrev() == null) {
            field = getParent().getField(getToken().getString());
        }
    }

    public Field getField() {
        return field;
    }

    @Override
    public void setNext(LineCall next) {
        super.setNext(next);
        if (next.getType() == Type.Value || getNext().getType() == Type.Struct || getNext().getType() == Type.Function) {
            getContext().error(next.getToken(), Error.lineUnexpectedCall);
        }
    }

    @Override
    public String toString() {
        return ""+getToken();
    }
}

File: SimplexApp/simplex/lang/parser/logic/line/call/CallField.js
const LineCall = require("simplex/lang/parser/logic/line/call/LineCall.js");
var Error;

class CallField extends LineCall {
    constructor(parent, token) {
        super(parent, token, LineCall.Field);
        this.field = null;
    }

    load() {
        if (this.getPrev() === null) {
            this.field = this.getParent().getField(this.getToken().getString());
        }
    }

    getField() {
        return this.field;
    }

    setNext(next) {
        super.setNext(next);
        if (next.getType() === LineCall.Value || this.getNext().getType() === LineCall.Struct || this.getNext().getType() === LineCall.Function) {
            this.getParent().error(next.getToken(), Error.lineUnexpectedCall);
        }
    }

    toString() {
        return '' + this.getToken();
    }
}

module.exports = CallField;
Error = require("simplex/lang/parser/logic/error/Error.js");